patchwork/scheduling.py

```diff
@@ -1,16 +1,23 @@
-import logging
+from collections import deque
+from typing import Any, Deque, Dict, Iterable, List, Optional, Set, Tuple, Union
 
-from collections import defaultdict, deque
-from textwrap import indent
-from typing import Callable, DefaultDict, Deque, Dict, Generator, List, Optional, Set, Tuple, Union
-
-from .actions import Action, PredictableAction, UnpredictableAction
+from .actions import Action
 from .context import Context
 from .datastore import Address, Datastore, TransactionAccumulator
-from .hypertext import Hypertext, Workspace
+from .hypertext import Workspace
 
 from .text_manipulation import insert_raw_hypertext, make_link_texts
 
+
+# Credits: Adapted from first_true in itertools docs.
+def next_truthy(iterable: Iterable[Any], default: Any) -> Any:
```

Comment:
Use [generics](https://mypy.readthedocs.io/en/latest/generics.html) instead of any?

Reply:
I thought about it, but `next_truthy` is analogous to `next` and `next` doesn't use generics. So I chose consistency over rigorous typing.

Reply:
Looks to me like `next` is using generics, see [here](https://github.com/python/typeshed/blob/master/stdlib/3/builtins.pyi#L859).

Reply:
Dang, I was reasoning "doesn't typecheck → doesn't compile", so "compiles → typechecks → next doesn't use generics". But of course, Python doesn't work like that. I will fix it tomorrow.

Reply:
Even if "doesn't typecheck → doesn't compile" was true, my reasoning would be wrong, because
```python
x = next_truthy(iter([1, 'banana']), 4)
```
typechecks with generics, too.